eo-runtime/src/main/eo/org/eolang/float.eo

```diff
@@ -29,21 +29,40 @@
 [] > float
   # Tests that $ = x
   [x] > eq
-    x > value!
+    x.as-bytes > x-as-bytes!
+    ^.as-bytes > self-as-bytes!
+    nan.as-bytes > nan-as-bytes!
+    0.0.as-bytes > pos-zero-as-bytes!
+    -0.0.as-bytes > neg-zero-as-bytes!
     if. > @
-      and.
+      or.
+        eq.
+          x-as-bytes
+          nan-as-bytes
         eq.
-          ^.as-bytes > self-as-bytes!
-          00-00-00-00-00-00-00-00 > zero-as-bytes!
+          self-as-bytes
+          nan-as-bytes
+      FALSE
+      if.
```

Comment:
@c71n93 I think this expression with `if` can be simplified with just `or.`

Reply:
@maxonfjvipon done